### PR TITLE
Expose cache and fix bool params

### PR DESCRIPTION
**Pull request by fbad**

### CODE REVIEW

**Review by joshuafleck — APPROVED** [review]

File: lib/avro_turf/connection_wrapper.rb [hunk 1]
```diff
@@ -37,8 +37,8 @@ def initialize(
     @client_cert_data = client_cert_data
     @client_key_data = client_key_data
     @connection_pool_size = connection_pool_size || CONNECTION_POOL_SIZE
-    @tcp_nodelay = tcp_nodelay || TCP_NODELAY
-    @persistent_connection = persistent_connection || PERSISTENT_CONNECTION
+    @tcp_nodelay = tcp_nodelay.nil? ? TCP_NODELAY : tcp_nodelay
```
Comment (mijoharas) [inline lib/avro_turf/connection_wrapper.rb:40]: not related, but any reason for this? the former is a pretty common rails pattern. (was there something else falsey that we might have been passing in here?)

Comment (fbad) [reply lib/avro_turf/connection_wrapper.rb:40]: if tcp_nodelay was set to false we were setting it to true anyway

Comment (fbad) [reply lib/avro_turf/connection_wrapper.rb:40]: false || true

Comment (mijoharas) [reply lib/avro_turf/connection_wrapper.rb:40]: got it, so there were falsey things being passed in. :+1:

Comment (fbad) [reply lib/avro_turf/connection_wrapper.rb:40]: not in our case but potentially yes

**Pull request merged**